Explorer/Assets/DCL/Multiplayer/Connections/Archipelago/Rooms/ArchipelagoIslandRoom.cs

```diff
@@ -128,6 +130,12 @@ private async UniTask<LightResult<string>> WelcomePeerIdAsync(string adapterUrl,
                 return LightResult<string>.FAILURE;
             }
 
+            if (!handshakePayloadValidationRegex.IsMatch(messageForSignResult.Result))
```

Comment:
We should avoid regex due to unnessecary allocations, will this work and not allocate every time we connect?

`messageForSignResult.StartsWith("dcl-") && messageForSignResult.AsSpan(4).IndexOfAny(':', ' ') == -1;`